src/Artsy/Analytics/trackPageViewWrapper.tsx

```diff
@@ -1,19 +1,47 @@
+import { isEqual } from "lodash"
 import React from "react"
+import { get } from "Utils/get"
 import { trackPageView } from "./trackPageView"
 
+interface Params {
+  [x: string]: string
+}
+
+interface State {
+  params: Params
+}
+
 export function trackPageViewWrapper<T>(
   Component: React.ComponentType<T>
 ): React.ComponentType<T> {
-  return class TrackPageView extends React.Component<T> {
+  return class TrackPageView extends React.Component<T, State> {
     static displayName = `TrackPageView(${Component.displayName ||
       Component.name ||
       "unknown"})`
+
+    constructor(props) {
+      super(props)
+
+      this.state = {
+        params: this.routeParams,
+      }
+    }
+
+    // TODO: Type props properly, can have optional `match`
```

Comment:
I spent a while trying to get the generic props to extend an interface that optionally included a `match` property, but couldn't quite get it. Since we're using `get` anyway, I just rolled with it.